src/pow.cpp

```diff
@@ -183,6 +262,40 @@ unsigned int CalculateNextWorkRequired(const CBlockIndex* pindexLast, const Cons
     return bnNew.GetCompact();
 }
 
+uint32_t CalculateNextWorkRequiredAux(const CBlockIndex* pindexLast, const Consensus::Params& params)
+{
+    arith_uint256 bnPowLimit = UintToArith256(params.aux_pow_limit);
+
+    // The very first block to have an auxiliary proof-of-work starts with a
+    // minimal difficulty.
+    if (!pindexLast || pindexLast->m_aux_pow.IsNull()) {
+        return bnPowLimit.GetCompact();
+    }
+    if (params.fPowNoRetargeting) {
+        return pindexLast->m_aux_pow.m_commit_bits;
+    }
+
+    std::pair<int64_t, int64_t> adjustment_factor;
+
+    adjustment_factor = GetFilteredAdjustmentFactorAux(pindexLast, params);
+
+    assert(adjustment_factor.first >= 0);
+    assert(adjustment_factor.second > 0);
+
+    // Retarget
+    arith_uint256 bnNew;
+    bnNew.SetCompact(pindexLast->m_aux_pow.m_commit_bits);
+    arith_uint320 bnTmp(bnNew);
+    bnTmp *= arith_uint320(static_cast<uint64_t>(adjustment_factor.second));
+    bnTmp /= arith_uint320(static_cast<uint64_t>(adjustment_factor.first));
+    assert(bnTmp.TruncateTo256(bnNew));
+
+    if (bnNew > bnPowLimit)
+        bnNew = bnPowLimit;
```

Comment:
Style nit, `{}` for multi-line if clause.

Alternative:
```C++
return (bnNew > bnPowLimit ? bnPowLimit : bnNew).GetCompact();
```

Reply:
Another, albeit a bit crazy, alternative:
```C++
return (bnTmp.TruncateTo256(bnNew) && bnNew < bnPowLimit ? bnNew : bnPowLimit).GetCompact();
```
which would replace from L#291.

Reply:
Went with the first option.